### PR TITLE
Enable the exploration of bib entries' relations (cited by and citing) 

### PR DESCRIPTION
**Pull request by HoussemNasri**
A continuation to https://github.com/JabRef/jabref/pull/7160/.

Closes https://github.com/JabRef/jabref/issues/6187

### Known Limitations
- Related entries cached in-memory; should be on disk.
- Relations are fetched exclusively based on the entry's DOI (SemantciScholar allows the usage of other types of ids .e.g. URL, ARXIV, PMID, etc.
- No pagination

### Screenshots

image

### Mandatory checks
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by the-dino-girl** [comment]
Can I ask: is this the same as this request I made some time ago?
https://github.com/JabRef/jabref/issues/10200

**Comment by HoussemNasri** [comment]
@the-dino-girl It's similar to that, but the references we get from a public web API (similar to how fetchers work). It's also possible to get bib entries that cites the selected entry. It's under the "cited by" section.

**Comment by the-dino-girl** [comment]
Can't wait for this feature!
I will extensively test it and report bugs (if any)

😍😍😍

**Comment by koppor** [comment]
> Can't wait for this feature! I will extensively test it and report bugs (if any)

You can download the current build at https://builds.jabref.org/pull/10324/merge and report issues here.

(This is what the comment https://github.com/JabRef/jabref/pull/10324#issuecomment-1705734369 means -- grab the binary and play around with it)

**Review by Siedlerchr — DISMISSED** [review]

**Comment by Siedlerchr** [comment]
Cool feature! Need to try this!

File: src/main/java/org/jabref/gui/entryeditor/citationrelationtab/BibEntryRelationsCache.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.identifier.DOI;
+
+public class BibEntryRelationsCache {
```
[the remark this answers is not in the document]
Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/entryeditor/citationrelationtab/BibEntryRelationsCache.java:11]: Yes, like the journal abbreviation stuff. Write the stuff to a db int he user's app data dir

File: src/main/java/org/jabref/gui/entryeditor/citationrelationtab/CitationRelationsTab.java [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+                preferencesService,
+                fileUpdateMonitor,
+                undoManager,
+                stateManager,
+                dialogService,
+                Globals.TASK_EXECUTOR);
+        importHandler.importEntries(entries);
+
+        dialogService.notify(Localization.lang("Number of entries successfully imported") + ": " + entriesToImport.size());
+    }
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/entryeditor/citationrelationtab/CitationRelationsTab.java:414]: I have the strong suspicion, that some of this code belongs into a ViewModel

Comment (HoussemNasri) [reply src/main/java/org/jabref/gui/entryeditor/citationrelationtab/CitationRelationsTab.java:414]: It surely does, but I thought we release the feature now letting users try it out and do some refactoring while working on the known limitations in later PRs.

Comment (calixtus) [reply src/main/java/org/jabref/gui/entryeditor/citationrelationtab/CitationRelationsTab.java:414]: ok, im fine with that. let ua just not forget about this, maybe an issue in the maintainers focus or sthg

**Pull request merged**